### PR TITLE
Prefilter graphs in the uploader

### PR DESCRIPTION
**Pull request by davidsoergel**
There is no point in uploading GraphDef data that won't be displayed anyway.  In particular, the TensorBoard frontend filters out node attributes larger that 1024 bytes, since it has no good way to present those.  So we may as well filter those out before upload to TensorBoard.dev, so as not to waste bandwidth, storage, and read-time processing.

### CODE REVIEW

File: tensorboard/dataclass_compat.py [hunk 1]
```diff
@@ -37,7 +40,7 @@
 from tensorboard.util import tensor_util
 
 
-def migrate_event(event):
+def migrate_event(event, experimental_filter_graph=False):
```
Comment (caisq) [inline tensorboard/dataclass_compat.py:43]: Please add doc string for the new kwarg.

File: tensorboard/backend/process_graph.py [hunk 2]
```diff
@@ -46,6 +46,9 @@ def prepare_graph_for_ui(
       ValueError: If `large_attrs_key is None` while `limit_attr_size != None`.
       ValueError: If `limit_attr_size` is defined, but <= 0.
     """
+    # TODO(@davidsoergel): detect whether a graph has been filtered already
```
Comment (caisq) [inline tensorboard/backend/process_graph.py:49]: To make this TODO item clearer, I believe you can say in addition: "if it is already filtered, return immediately".

File: tensorboard/dataclass_compat_test.py [hunk 3]
```diff
@@ -212,6 +215,49 @@ def test_graph_def(self):
 
         self.assertProtoEquals(graph_def, new_graph_def)
 
+    def test_graph_def_experimental_filter_graph(self):
+        # Create a `GraphDef` and write it to disk as an event.
+        logdir = self.get_temp_dir()
```
Comment (caisq) [inline tensorboard/dataclass_compat_test.py:220]: Question about testing strategy: is it really necessary to create a logdir and a writer? If your intention here is to simply test `self._migrate_event()` (and ultimately `dataclass_compat.migrate_event()`), you can just construct a `Event`, pass it to that method and check the result, right?

File: tensorboard/backend/process_graph.py (hunk 2 above)
Comment (davidsoergel) [reply tensorboard/backend/process_graph.py:49]: Done.

File: tensorboard/dataclass_compat.py (hunk 1 above)
Comment (davidsoergel) [reply tensorboard/dataclass_compat.py:43]: Oops, done.

File: tensorboard/dataclass_compat_test.py (hunk 3 above)
Comment (davidsoergel) [reply tensorboard/dataclass_compat_test.py]: Oh good point.  I had pasted that from the test above but here the logdir part is superfluous.  (There too, maybe, but that's out of scope).  Thanks!

**Pull request merged**